pkg/virtctl/expose/expose.go

```diff
@@ -0,0 +1,213 @@
+// executing the "expose" command
+func (o *Command) RunE(cmd *cobra.Command, args []string) error {
+	// first argument is type of VM: VM, offline VM or replica set VM
+	vmType := strings.ToLower(args[0])
+	// second argument must be name of the VM
+	vmName := args[1]
+
+	// these are used to convert the flag values into service spec values
+	var protocol v1.Protocol
+	var targetPort intstr.IntOrString
+	var serviceType v1.ServiceType
+
+	// convert from integer to the IntOrString type
+	targetPort = intstr.FromInt(intTargetPort)
+
+	// convert from string to the protocol enum
+	switch strProtocol {
+	case "TCP":
+		protocol = v1.ProtocolTCP
+	case "UDP":
+		protocol = v1.ProtocolUDP
+	default:
+		return fmt.Errorf("Unknown protocol: %s", strProtocol)
+	}
+
+	// convert from string to the service type enum
+	switch strServiceType {
+	case "ClusterIP":
+		serviceType = v1.ServiceTypeClusterIP
+	case "NodePort":
+		serviceType = v1.ServiceTypeNodePort
+	case "LoadBalancer":
+		serviceType = v1.ServiceTypeLoadBalancer
+	case "ExternalName":
+		return fmt.Errorf("Type: %s not supported", strServiceType)
+	default:
+		return fmt.Errorf("Unknown service type: %s", strServiceType)
+	}
+
+	// get the namespace
+	namespace, _, err := o.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+
+	// get the client
+	virtClient, err := kubecli.GetKubevirtClientFromClientConfig(o.clientConfig)
+	if err != nil {
+		return fmt.Errorf("Cannot obtain KubeVirt client: %v", err)
+	}
+
+	// does a plain quorum read from the apiserver
+	options := k8smetav1.GetOptions{}
+
+	switch vmType {
+	case "vm", "vms", "virtualmachine", "virtualmachines":
+		// get the VM
+		_, err := virtClient.VM(namespace).Get(vmName, options)
+		if err != nil {
+			return fmt.Errorf("Error fetching VirtualMachine: %v", err)
+		}
+	case "ovm", "ovms", "offlinevirtualmachine", "offlinevirtualmachines":
+		// get the offline VM
+		_, err := virtClient.OfflineVirtualMachine(namespace).Get(vmName, &options)
+		if err != nil {
+			return fmt.Errorf("Error fetching OfflineVirtualMachine: %v", err)
+		}
+	case "vmrs", "vmrss", "virtualmachinereplicaset", "virtualmachinereplicasets":
+		// get the VM replica set
+		_, err := virtClient.ReplicaSet(namespace).Get(vmName, options)
+		if err != nil {
+			return fmt.Errorf("Error fetching VirtualMachine ReplicaSet: %v", err)
+		}
+		// in case of replica set we take the label from the replica set template
+        // same as the original vmName
+		//vmName = vmrs.Spec.Template.ObjectMeta.Labels["kubevirt.io/vmReplicaSet"]
+	default:
+		return fmt.Errorf("Unsupported resource type: %s", vmType)
+	}
+
+	// actually create the service
+	service := &v1.Service{
+		ObjectMeta: k8smetav1.ObjectMeta{
+			Name: serviceName,
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{Name: portName, Protocol: protocol, Port: port, TargetPort: targetPort, NodePort: nodePort},
+			},
+			Selector:       map[string]string{"kubevirt.io/domain": vmName},
```

Comment:
Checked the `kubectl expose` code. I think we should align and  not allowing exposing the object if not labels are present at all. I was not sure about how to behave in this case before, but I would align with the behaviour they have, to avoid confusions.

Reply:
still needs to be addressed.